### DISCUSSION TITLE
copilot.vim: Copilot: Disabled for filetype=markdown by internal default

### DISCUSSION THREAD

**Discussion by unphased** [community/community#29977] [Copilot Conversations]
I tried running `:Copilot enable` but it had no effect? `:set ft=txt` did work. But that is not ideal.

**Comment by mathomp4** [comment]
I'm wondering if this is what I'm seeing? When I run copilot with a Python file, everything seems to work in vim. I type, a suggestion is made, I press <Tab> and I get the suggestion.

But when I edit a Markdown file, I get a suggestion, but <Tab> just...tabs. It doesn't seem to complete the suggestion. (Not that I'd expect Copilot to work well for a README...but it actually does guess pretty well.)

**Reply by mathomp4** [reply]
Note: I can say that GitHub Copilot in VS Code does seem to work with Markdown. I get suggestions and it accepts them. Not the weird "Ghost and No Tab" I see in vim

**Reply by mathomp4** [reply]
And I can say that it doesn't seem to be a Vim v Neovim thing. Both aren't happy.

**Comment by unphased** [comment]
I do not even get a suggestion. I looked at the source and it seems like it just forces it off for markdown so i work around it by setting the filetype to txt, but that's silly because it removes the markdown syntax highlighting.

**Comment by txgruppi** [comment]
I had the same issue, I want to `autoload/copilot.vim` and removed `markdown` from `s:filetype_defaults`.

 It works fine now. (until I updated the plugin code, I know)

**Comment by unphased** [comment]
Yes but requiring us to manually mod the plugin code is not an acceptable solution. Please allow us to set a global variable to adjust the list of disabled filetypes.

As well as enable a manual `:Copilot enable` to override this (Though that is also not a long term solution, just a more sensible behavior)

**Reply by mathomp4** [reply]
Per `:help Copilot`, does:

```vim
let g:copilot_filetypes = {
   \ 'markdown': v:true,
   \ }
```
re-enable for markdown? I'd have to stare through the code to see which setting has priority...

Edited to make it right as @unphased figured out!

**Reply by unphased** [reply]
It does not work. Thank you for the suggestion though.

**Reply by unphased** [reply]
OK it seems like it works if: 

- using `v:true` instead of `true`

That does the trick alright! Excellent!

**Reply by mathomp4** [reply]
Oh, bad type by me. Per `:help copilot`:

```vim
                                                g:copilot_filetypes
g:copilot_filetypes     A dictionary mapping file types to their enabled
                        status.  Most file types are enabled by default, so
                        generally this is used for opting out.

                        let g:copilot_filetypes = {
                              \ 'xml': v:false,
                              \ }

                        Disabling all file types can be done by setting the
                        special key "*".  File types can then be turned back
                        on individually.

                        let g:copilot_filetypes = {
                              \ '*': v:false,
                              \ 'python': v:true,
                              \ }
```

**Reply by MorganBergen** [reply]
do we modify the copilot.txt file directly to set to true for all file types?

**Reply by MorganBergen** [reply]
<img width="1440" alt="Screenshot 2022-11-02 at 9 47 04 AM" src="https://user-images.githubusercontent.com/65584733/199520962-ce37a0d7-1583-4557-ad23-919d6c7e14db.png">

Im new to nevoid, where do we edit and set for all

**Reply by mathomp4** [reply]
@MorganBergen I can help out, but I'm not sure what nevoid is. Is it a vim/neovim clone?

**Reply by nhussein11** [reply]
> do we modify the copilot.txt file directly to set to true for all file types?

Hey! You don't need to modify the copilot.txt file directly, you just need to set up the following: https://github.com/orgs/community/discussions/38074#discussioncomment-4049799

If you are using some vim framework, I suggest you to look for the equivalent to .vimrc there and modify it.
